addons/web/static/tests/views/kanban_view_tests.js

```diff
@@ -11809,4 +11809,68 @@ QUnit.module("Views", (hooks) => {
         assert.deepEqual(getCardTexts(0), ["blip-4"]);
         assert.deepEqual(getCardTexts(1), ["yop10", "blip9", "gnap17", "cba13"]);
     });
+
+    QUnit.test("no demo data when all groups are folded then one is unfolded", async (assert) => {
+        await makeView({
+            type: "kanban",
+            resModel: "partner",
+            serverData,
+            arch: /* xml */ `
+                <kanban sample="1">
+                    <templates>
+                        <div t-name="kanban-box">
+                            <field name="id"/>
+                        </div>
+                    </templates>
+                </kanban>`,
+            groupBy: ["product_id"],
+            async mockRPC(route, args, performRPC) {
+                if (args.method === "web_read_group") {
+                    const result = await performRPC(route, args);
+                    for (const group of result.groups) {
+                        group.__fold = true;
+                    }
+                    return result;
+                }
+                return performRPC(route, args);
+            },
+        });
+
+        const groupHandle = target.querySelector(".o_kanban_group");
+        await click(groupHandle);
+
+        assert.containsN(target, ".o_kanban_record", 2); // instead of the the many in the demo data
```

Comment:
this is too weak, you don't know how much records are generated in the **sample** (not demo) data. When the sample data are activated, there's a className on the kanban view I think. It would be more reliable. Also, assert that groups are correctly folded before